lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -509,6 +538,23 @@ abstract class PrimaryRoleManager {
   /// the object.
   @mustCallSuper
   void update() {
+    final Map<Role, RoleManagerFactory>? secondaryRoleFactories = _secondaryRoleFactories;
+    if (secondaryRoleFactories != null && secondaryRoleFactories.isNotEmpty) {
+      secondaryRoleFactories.forEach((Role role, RoleManagerFactory roleFactory) {
+        final RoleManager? secondaryRoleManager = roleFactory(semanticsObject);
+        if (secondaryRoleManager != null) {
```

Comment:
This for loop looks a little bit weird to me. Wouldn't it be better that this knows what `availableSecondaryRoles` can be used, and the `semanticsObject` knows what `configuredSecondaryRoles` it has?

That way instead of this loop checking on every roleFactory if it can support the semanticsObject, an intersection would narrow down the actually needed Roles (which could have the roleFactory associated to the Role an immutable `const Map<Role, RoleManagerFactory>`, or whatever?)